src/ex1/script.js

```diff
@@ -0,0 +1,129 @@
+const addTaskButton = document.querySelector(".add-task");
+const taskInput = document.querySelector("#task-text");
+const allTasksContainer = document.querySelector(".all-tasks-container");
+const pendingTasksText = document.querySelector(".pending-tasks");
+let pendingTasks = 0;
+const clearButton = document.querySelector(".clear-all");
+const sortButton = document.querySelector(".name-sort");
+
+// Event trigger when user clicks the add task button
+addTaskButton.addEventListener("click", () => {
+   addTask();
+});
+
+// Same as the button add task but only when user press "Enter"
+taskInput.addEventListener("keypress", (e) => {
+   if (e.key === "Enter") {
+      addTask();
+   }
+});
+
+// Clears all tasks when clear all button is clicked
+clearButton.onclick = () => {
+   document
+      .querySelectorAll(".task-container")
+      .forEach((task) => task.remove());
+   pendingTasksUpdate(0);
+};
+
+// Sorts the tasks containers by name
+sortButton.onclick = () => {
+   const taskContainers = [...document.querySelectorAll(".task-container")];
+   taskContainers.sort((a, b) =>
+      a.children[0].innerText.localeCompare(b.children[0].innerText)
+   );
+   taskContainers.map((container) => container.remove());
+   taskContainers.map((container) => allTasksContainer.appendChild(container));
+};
+
+// Adds the user task input to the tasks list
+function addTask() {
+   // Checks for blank spaces
+   if (taskInput.value.trim().length === 0) {
+      alert("Please fill in a task");
+      taskInput.value = "";
+   } else {
+      const taskContainer = createTaskContainer();
+      const newTask = createNewTask(taskContainer);
+      const deleteTask = createDeleteTaskButton(taskContainer);
+      addHoverReveal(taskContainer, deleteTask, newTask);
+      createTaskAnimation(newTask);
+      taskInput.value = "";
+      pendingTasksUpdate("+");
+   }
+}
+
+// Creates container which holds task box and delete button
+function createTaskContainer() {
+   const taskContainer = document.createElement("div");
+   taskContainer.classList.add("task-container");
+   allTasksContainer.append(taskContainer);
+
+   return taskContainer;
+}
+
+// Creates task box and click alert event to it
+function createNewTask(taskContainer) {
+   const newTask = document.createElement("div");
+   newTask.textContent = taskInput.value;
```

Comment:
consider: I would extract `taskInput.value` to a function to encapsulate the "logic" (for example you can use the `trim` logic in that function)

Reply:
I agree but here I create a div element first. If I find that the user input was blank after then I have to delete this element. Maybe in the future exercises when I check for more complicated logic then it will be more suitable?
I think for the black space it's better to check first when the user clicks the "add" button.